TravelBook/app/src/main/java/com/example/travelbook/profile/viewModels/ProfileViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.travelbook.profile.viewModels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.NavHostController
+import com.example.travelbook.User
+import com.example.travelbook.UserDataSource
+import com.example.travelbook.navigation.models.NavigationItem
+import com.example.travelbook.signIn.AuthRepo
+import com.example.travelbook.signIn.viewModels.NewSignInViewModel
+
+class ProfileViewModel (
+    private val navigationController: NavHostController,
+    private val userDataSource: UserDataSource,
+    private val repository: AuthRepo
+) : ViewModel() {
+    fun onSignOutClicked() {
+        repository.signOut()
+        userDataSource.deleteUser()
+        navigationController.navigate(NavigationItem.SignIn.route)
```

Comment:
might be worth ensuring that sign out was successful (check fb current user is null) and deleteUser was also successful (nothing saved in sharedPrefs) before navigating back to the sign in page?

Reply:
updated pr with these changes